sshtunnel.cpp

```diff
@@ -363,6 +372,80 @@ bool SshTunnel::prompt_password()
     return true;
 }
 
+bool SshTunnel::interactive()
+{
+    AppSettings settings;
+
+    int result = ssh_userauth_kbdint(m_ssh, nullptr, nullptr);
+    while (result == SSH_AUTH_INFO) {
+        const char *name, *instruction;
+        int nprompts;
+
+        name = ssh_userauth_kbdint_getname(m_ssh);
+        instruction = ssh_userauth_kbdint_getinstruction(m_ssh);
+        nprompts = ssh_userauth_kbdint_getnprompts(m_ssh);
+
+        Gtk::Dialog dialog(name, m_parent);
+        dialog.add_button("_Cancel", Gtk::RESPONSE_CANCEL);
+        dialog.add_button("_Ok", Gtk::RESPONSE_OK);
+        dialog.set_default_response(Gtk::RESPONSE_OK);
+
+        auto *grid = Gtk::manage(new Gtk::Grid);
+        grid->set_row_spacing(10);
+        grid->set_column_spacing(5);
+        grid->set_border_width(5);
+
+        Gtk::Label *instruction_label = Gtk::manage(new Gtk::Label(instruction));
+        instruction_label->set_alignment(Gtk::ALIGN_START, Gtk::ALIGN_START);
+        grid->attach(*instruction_label, 0, 0, 2, 1);
+
+        for (int i=0; i<nprompts; i++) {
+            char echo;
+            const char *prompt = ssh_userauth_kbdint_getprompt(m_ssh, i, &echo);
+
+            Gtk::Label *prompt_label = Gtk::manage(new Gtk::Label(prompt));
+            prompt_label->set_alignment(Gtk::ALIGN_START, Gtk::ALIGN_START);
+
+            Gtk::Entry *input_entry = Gtk::manage(new Gtk::Entry);
+            input_entry->set_activates_default(true);
+            input_entry->set_visibility(echo);
+
+            grid->attach(*prompt_label, 0, (i + 1), 1, 1);
+            grid->attach(*input_entry, 1, (i + 1), 1, 1);
+        }
+
+        if (nprompts > 0) {
+            auto vbox = dialog.get_child();
+            dynamic_cast<Gtk::Container *>(vbox)->add(*grid);
+            dialog.show_all();
+            int response = dialog.run();
+            dialog.hide();
+
+            if (response != Gtk::RESPONSE_OK)
+                return false;
+        }
+
+        for (int i=0; i<nprompts; i++) {
+            Gtk::Entry* input_entry = dynamic_cast<Gtk::Entry*>(grid->get_child_at(1, i + 1));
+            if (input_entry) {
+                ssh_userauth_kbdint_setanswer(m_ssh, i, input_entry->get_text().c_str() );
```

Comment:
```suggestion
                ssh_userauth_kbdint_setanswer(m_ssh, i, input_entry->get_text().c_str());
```

Reply:
Updated and did some small adjustments:
- assume all auth methods if none are marked as available
- show interactive dialog even with no prompts if there is an instruction